### PR TITLE
ArchiSteamFarm: 5.2.8.3 -> 5.3.1.2

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/ArchiSteamFarm/update.sh [hunk 1]
```diff
@@ -1,10 +1,8 @@
 #!/usr/bin/env nix-shell
-#!nix-shell -I nixpkgs=../../../.. -i bash -p curl gnused jq common-updater-scripts
```
Comment (IvarWithoutBones) [inline pkgs/applications/misc/ArchiSteamFarm/update.sh:2]: Using the local nixpkgs checkout seems valuable, in case that contains necessary updates not present in the channels yet.

 I also recall the maintainer script having trouble when not executed from nixpkgs root, but I'm not entirely sure about that.

File: pkgs/applications/misc/ArchiSteamFarm/update.sh [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
+#!nix-shell -i bash -p curl gnused jq common-updater-scripts
 set -euo pipefail
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
-deps_file="$(realpath ./deps.nix)"
```
Comment (IvarWithoutBones) [inline pkgs/applications/misc/ArchiSteamFarm/update.sh:6]: Why was this removed? The call to the nuget deps generation script below is broken now.

File: pkgs/applications/misc/ArchiSteamFarm/update.sh [hunk 3]
```diff
@@ -15,15 +13,14 @@ if [[ "$new_version" == "$old_version" ]]; then
...
 if [[ "${1:-}" != "--deps-only" ]]; then
     update-source-version ArchiSteamFarm "$new_version"
 fi
 
 $(nix-build -A ArchiSteamFarm.fetch-deps --no-out-link) "$deps_file"
 
-cd "$asf_path"
-./web-ui/update.sh
+cd "$asf_path"/web-ui/
+./update.sh
```
Comment (IvarWithoutBones) [inline pkgs/applications/misc/ArchiSteamFarm/update.sh:26]: I think it would look a lot more intuative if we could use `exec "$asf_path"/web-ui/update.sh`, not sure if the script depends on the working directory though.

**Review by IvarWithoutBones — COMMENTED** [review]
I currently do not have time to test this but changes to `buildDotnetModule` look reasonable, I just have a few comments on the update scripts.

File: pkgs/applications/misc/ArchiSteamFarm/update.sh (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/ArchiSteamFarm/update.sh]: The problem is that you then always need to cd into the directory where the update script is.
